xmovie/core.py

```diff
@@ -358,12 +370,66 @@ def save_frames(self, odir, progress=False):
         elif ~tqdm_avail and progress:
             warnings.warn("Cant show progess bar at this point. Install tqdm")
 
-        for fi in frame_range:
-            fig, ax, pp = self.render_frame(fi)
-            frame_save(
-                fig, fi, odir=odir, frame_pattern=self.frame_pattern, dpi=self.dpi
+        for timestep in frame_range:
+            fig, ax, pp = self.render_single_frame(timestep)
+            save_single_frame(
+                fig, timestep, odir=odir, frame_pattern=self.frame_pattern, dpi=self.dpi
             )
 
+
+
+    def save_frames_parallel(self, odir, progress=False, 
+                             parallel_compute_kwargs=dict()):
+        '''
+        Saves all frames in parallel using dask.map_blocks.
+
+        Parameters
+        ----------
+        odir : path
+            path to output directory
+        progress : Bool
+            Whether or not to show a progress bar.
+        parallel_compute_kwargs : dict
+            Keyword arguments to pass t dask's `compute()` function.
+
+        '''
+        import numpy as np
+        import dask.array as darray
+
+        da = self.data
+        framedim = self.framedim
+        
+        da = da.chunk({self.framedim : 1}) # DataArray needs to have length-1 chunks in framedim
+        chunk_dims = [ dim for dim in da.dims if dim != framedim ] # Create dimensions for each chunk
+               
+
+        total_time = da[framedim]
+        def create_render_save_single_frame2(xr_array, framedim):
+            time_of_chunk = xr_array[framedim]
+            timestep = abs(total_time - time_of_chunk[0]).argmin().item() # get index of chunk in framedim
+
+            fig, ax, pp = self.render_single_frame(timestep)
+            save_single_frame(fig, timestep, odir=odir, frame_pattern=self.frame_pattern, dpi=self.dpi)
+
+            return time_of_chunk
+
+        mapped_save_and_close_frames = da.map_blocks(func=create_render_save_single_frame2,
+                                                     args=(framedim,),
+                                                     template=xr.ones_like(da[framedim]).chunk({framedim:1}),
```

Comment:
Same as above. I think we should expect the user to chunk along the framedim. This could also improve performance.

Reply:
Ah sorry, I think I did not read this correctly. Is this slower than creating a dask array explicitly, with something like this:
```
template=xr.Dataarray(da.ones(len(da[framedim], chunks=(1,)))
```

Reply:
If I understood your question correctly, they are similar in time. The version I used has the advantage of being shorter and easier to read, but it's a bit slower. Although the cost of this operation is certainly negligible compared to the time saving takes.

```python
In [24]: %timeit xr.ones_like(da["time"]).chunk({"time":1})
955 µs ± 11.6 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

In [25]: %timeit template=xr.DataArray(dask.array.ones(len(da["time"]), chunks=(1,)))
696 µs ± 3.19 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
```

Since the time different is (IMO) negligible, I prefer to keep it this way. But let me know if you disagree.

Reply:
Fine with me, unless the xarray wunderkind @dcherian has an even more readable suggestion!